src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs

```diff
@@ -50,22 +52,41 @@ internal sealed class ClassEvaluator : MemberEvaluator {
                 EvaluateInnerClasses(_classDef);
 
                 _class = classInfo;
+
                 // Set bases to the class.
                 var bases = new List<IPythonType>();
                 foreach (var a in _classDef.Bases.Where(a => string.IsNullOrEmpty(a.Name))) {
-                    // We cheat slightly and treat base classes as annotations.
-                    var b = Eval.GetTypeFromAnnotation(a.Expression);
-                    if (b != null) {
-                        var t = b.GetPythonType();
-                        bases.Add(t);
-                        t.AddReference(Eval.GetLocationOfName(a.Expression));
+                    // A base can either be the name of a class declared in the current module or a class declared in some other module
+                    switch (a.Expression) {
```

Comment:
The old logic didn't check between Name/Member expressions, and the new logic does different work; what does the old code not do that the new code needs to do?

Reply:
The old code would add to bases everything that was put into the class

For example 

```
class C(5, 6):
  def test(self):
     pass

``` was valid.

I wanted to make sure that the only thing added was python classes, so I distinguished between NameExpressions of other classes and classes declared in other modules.

Reply:
Makes sense, though you may want to go look at those failing tests. I think they're to do with bases which come from specialized things (like generics, or specifically specialized types).